File: src/main/java/tmi/recipe/parser/VariableReactorParser.java
package tmi.recipe.parser;

import arc.struct.Seq;
import mindustry.content.Blocks;
import mindustry.world.Block;
import mindustry.world.blocks.power.ConsumeGenerator;
import mindustry.world.blocks.power.VariableReactor;
import mindustry.world.meta.StatUnit;
import tmi.recipe.Recipe;
import tmi.recipe.RecipeType;
import tmi.recipe.types.HeatMark;
import tmi.recipe.types.PowerMark;

public class VariableReactorParser extends ConsumerParser<VariableReactor>{
  {excludes.add(GeneratorParser.class);}

  @Override
  public boolean isTarget(Block content) {
    return content instanceof VariableReactor;
  }

  @Override
  public Seq<Recipe> parse(VariableReactor content) {
    Recipe res = new Recipe(RecipeType.generator)
        .setBlock(getWrap(content));

    registerCons(res, content.consumers);

    res.addProductionPresec(PowerMark.INSTANCE, content.powerProduction);

    if (content.maxHeat > 0){
      res.addProductionRaw(HeatMark.INSTANCE, content.maxHeat).setFloatFormat();
    }

    return Seq.with(res);
  }
}

File: src/main/kotlin/tmi/recipe/parser/VariableReactorParser.kt
package tmi.recipe.parser

import arc.struct.Seq
import mindustry.world.Block
import mindustry.world.blocks.power.VariableReactor
import tmi.recipe.Recipe
import tmi.recipe.RecipeParser
import tmi.recipe.types.RecipeItemType
import tmi.recipe.RecipeType
import tmi.recipe.types.HeatMark
import tmi.recipe.types.PowerMark

open class VariableReactorParser : ConsumerParser<VariableReactor>() {
  override val excludes: Seq<Class<out RecipeParser<*>>> = Seq.with(GeneratorParser::class.java)

  override fun isTarget(content: Block): Boolean {
    return content is VariableReactor
  }

  override fun parse(content: VariableReactor): Seq<Recipe> {
    val res = Recipe(
      recipeType = RecipeType.generator,
      ownerBlock = content.getWrap()
    )

    registerCons(res, *content.consumers)

    res.addProductionPersec(PowerMark, content.powerProduction)
      .setType(RecipeItemType.POWER)

    if (content.maxHeat > 0) {
      res.addProduction(HeatMark, content.maxHeat as Number)
        .setType(RecipeItemType.POWER)
        .floatFormat()
    }

    return Seq.with(res)
  }
}
